terraform/modules/front_door/waf_cloudfront.tf

```diff
@@ -1,6 +1,7 @@
-resource "aws_wafv2_web_acl" "this" {
+# When updating this cloudfront WAF acl, consider making the same change for the load balancer WAF acl, as they generally need to be the same

```

Comment:
So in general I expect us to have the same waf configuration for cloudfront and the load balancer (in terms of web acl and regex patterns). Cloudfront needs the us-east-1 provider and cloudfront scope, whereas load balancer needs eu-west-2 (our default) and regional scope.

Because of this I wanted to make the WAF a module and set these dynamically, but it turns out you can't set the provider dynamically, so I have had to essentially make two very similar files (but with these bits differing), and add a comment to top of both saying that you should consider updating both when changing any of the WAF configuration 😥. I don't like it but it's the best I could think of, if you know a better idea please let me know!

Reply:
What does it gain us to have (the same) waf set up on the load balancer as well as on cloudfront?

Reply:
My main thinking was our LB is public, so if a bad actor was able to send traffic directly there (bypassing cloudfront) then we'd have a similar level of protection at the LB (helping to reduce our overall attack surface on the app). There would be a few other layers to get through to achieve this though (sg's, headers etc), so it's not necessarily a particularly vulnerable area. Was also thinking about pen test outcomes.

I think this implementation implies doubling our WAF costs as it's now analysing the same stuff at two points, so there is a cost/benefit to ponder. I'm happy to remove the LB protection if you feel this trade-off isn't worth it?

Reply:
They would have to get hold of the header we use for preventing accessing the lb directly - which is our intended main source of protection here. 

Presumably as well as cost it also has some impact on performance, which isn't ideal. So I am generally reluctant to be applying it twice.

(N.B. I'm fairly certain delta only applies their waf rules to cloudfront, so fairly happy that it ought to pass a pen test)

Reply:
Yeah indeed, and I believe even with the header you'd need to get past the sg's (which checks the source has a cloudwatch prefix) 😀

Good point about the performance! I think I'll remove WAF from LB then all things considered. A lot of the PR will whittle down to just removing and updating the TODO comments rather than any implementation work, but it's good to have considered carefully, thanks 👍

Reply:
I've removed WAF protection on the LB and updated the code comments 😀👍